### DISCUSSION TITLE
merge TiMVT and TiFeatures project

### DISCUSSION THREAD

**Discussion by vincentsarago** [developmentseed/timvt#96] [Ideas]
TiMVT and TiFeatures shares a lot of in common and don't have huge difference in requirements (`morecantile/pyproj` for timvt and `pygeofilter/geojson-pydantic` for tifeatures). 

With @bitner we are thinking that we could merge both project into one project. I'm not quite sure we would want to have a unique python package for both features/mvt yet. This could be achieved by having namespace module (e.g tipostgis.features, tipostgis.mvt).

## Pro
- reduce code duplication
- easy mvt/feature application setup
- Factory option will let the user choose if it wants feature and mvt endpoint

## Cons
- add dependencies (morecantile/pyproj or pygeofilter/geojson-pydantic)
- might add complexity in the `CollectionLayer` class

-- 

Final application should have those endpoints:

- `{endpoint}/`: landing page 
- `{endpoint}/api.json`: openapi
- `{endpoint}/api.html`: swagger html page
- `{endpoint}/conformance`: OGC conformance
- `{endpoint}/collections`: List of collections (Table + Functions)
- `{endpoint}/collections/{collectionId}`: Collection metadata
- `{endpoint}/collections/{collectionId}/queryables`: Collection queryables
#### Features
- `{endpoint}/collections/{collectionId}/items`: Collection Features (with query options)
- `{endpoint}/collections/{collectionId}/items/{itemId}`: Collection Feature
#### MVT
- `{endpoint}/collections/{collectionId}/tiles/{tileMatrixSetId}/{tileMatrix}/{tileRow}/{tileCol}`: Collection MVT tiles
- `{endpoint}/collections/{collectionId}/tilejson.json`: Collection Tilejson document
- `{endpoint}/tileMatrixSets`: List of available TMS
- `{endpoint}/tileMatrixSets/{tileMatrixSetId}`:  Metadata about a specific TMS

maybe we should wait for https://ogcapi.ogc.org/tiles/

**Comment by rodrigoalmeida94** [comment]
I absolutely love this idea. In the project where I used tifeatures we ended up realizing close to the end that we would also want timvt for snappier visualization, so I think packaging everything in the same code base is really good. Could I suggest `tivector` as a name?

**Comment by vincentsarago** [comment]
FYI: https://github.com/developmentseed/tipg
